app/src/features/Preview/Preview.tsx

```diff
@@ -0,0 +1,215 @@
+    padding: theme.spacing(5),
+  },
+  table: {
+    borderRadius: `${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0 0`,
+    overflowX: "scroll",
+    border: `1px solid ${theme.palette.divider}`,
+    borderBottom: "none",
+    borderCollapse: "separate",
+    marginBottom: theme.spacing(3),
+  },
+  icon: {
+    width: theme.spacing(3),
+    height: theme.spacing(3),
+  },
+  iconFlame: {
+    fill: "#CC7831",
+  },
+  fhirIconCell: {
+    boxShadow: `inset -1px 0 0 ${theme.palette.divider}`,
+    position: "sticky",
+    left: 0,
+    padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
+    width: 25,
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  cells: {
+    whiteSpace: "nowrap",
+    boxShadow: `inset -1px 0 0 ${theme.palette.divider}`,
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  texts: {
+    textAlign: "center",
+  },
+  cellsTitle: {
+    background: theme.palette.background.paper,
+  },
+  rowBorder: {
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  link: {
+    alignSelf: "flex-start",
+    display: "flex",
+    alignItems: "center",
+  },
+  jsonViewer: {
+    padding: 0,
+    border: `1px solid ${theme.palette.divider}`,
+
+    borderRadius: theme.shape.borderRadius,
+    "& > div": {
+      padding: theme.spacing(3),
+      borderRadius: theme.shape.borderRadius,
+    },
+  },
+}));
+
+const Preview = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const { sourceId, mappingId, attributeId } = useParams<{
+    sourceId?: string;
+    mappingId?: string;
+    attributeId?: string;
+  }>();
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    {
+      id: mappingId ?? "",
+    },
+    {
+      skip: !Boolean(mappingId),
+    }
+  );
+  const [fhirInstance, setFhirInstance] = useState<
+    typeof patientJson | undefined
+  >(undefined);
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+
+  const FhirIconCell = () => (
+    <IconButton size="small" className={classes.icon}>
+      <Icon
+        icon={IconNames.FLAME}
+        iconSize={15}
+        className={classes.iconFlame}
+      />
+    </IconButton>
+  );
+
+  const handleFhirIconClick = (index: number) => () => {
+    const primarykey = mapping?.primary_key_table;
+    if (primarykey) {
+      const indexPrimaryKey = exploredData.fields.indexOf(primarykey);
+      console.log(exploredData.rows[index]?.[indexPrimaryKey]);
+      setFhirInstance(patientJson);
+    }
+  };
+
+  return (
+    <Container className={classes.container} maxWidth="xl">
+      <Link
+        className={classes.link}
+        variant="h5"
+        color="textSecondary"
+        to={
+          attributeId
+            ? `/sources/${sourceId}/mappings/${mappingId}/attributes/${attributeId}`
+            : `/sources/${sourceId}/mappings/${mappingId}`
+        }
+        gutterBottom
+      >
+        <ArrowBack />
+        {t("back")}
+      </Link>
```

Comment:
I suggest to use the `cancel` button set in `CreateMapping` component instead. You'll be able to keep consistency in style and more over, using an `onClick` event instead of the `to` props will allow you to call the `history.back` method, without worrying about both URLs

Reply:
You can keep the `back` translation of course

Reply:
I still have so many hooks to learn O.o